chromium_src/base/test/launcher/test_results_tracker.cc

```diff
@@ -0,0 +1,24 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+// Upstream deliberately avoids including the failure's details and stack trace
+// in the result XML for the xUnit format, so we need to add it explicitly to be
+// able to visualize the data directly from Jenkins for each failed test.
+#define TEST_RESULTS_TRACKER_ADD_FAILURE_DETAILS                             \
+  std::string failure_data;                                                  \
+  for (const TestResultPart& result_part : result.test_result_parts) {       \
+    failure_data += StringPrintf(                                            \
+        "Failure at %s (line %d):\n\n%s\n\n", result_part.file_name.c_str(), \
+        result_part.line_number, result_part.message.c_str());               \
+  }                                                                          \
+  fprintf(out_,                                                              \
+          "      <failure message=\"[  FAILED  ] %s.%s\" "                   \
+          "type=\"\"><![CDATA[%s]]></failure>\n",                            \
+          testsuite_name.c_str(), result.GetTestName().c_str(),              \
+          failure_data.c_str());
```

Comment:
in the original file another failure will be printed after this line. see: https://github.com/chromium/chromium/blob/8abdbcfdf4a0d114c9d27f6d2fe4ec1d1a419f24/base/test/launcher/test_results_tracker.cc#L150

maybe we should silence it?

Reply:
Yes, I have done it this way on purpose to minimize the patching surface (i.e. only adding one line vs adding & removing) after checking that Jenkins will only care about the first `<failure>` element being written.

We could of course still remove it, but if this is enough for our purpose I lean towards patching less, even if it's one line less, WDYT?

Reply:
sure, if it works without issues then we're good.